CMakeLists_files.cmake

```diff
@@ -183,6 +183,11 @@ list (APPEND MAIN_SOURCE_FILES
   opm/simulators/wells/WGState.cpp
   )
 
+if (HAVE_ECL_INPUT)
+  list (APPEND MAIN_SOURCE_FILES
+    opm/core/props/satfunc/SatfuncConsistencyChecks.cpp
```

Comment:
Maybe we should take this opportunity to replace core with something more descriptive like 'diagnostics' or such

Reply:
> Maybe we should take this opportunity to replace core with something more descriptive like 'diagnostics' or such

You mean, we'd have files named
```
opm/diagnostics/props/satfunc/ConsistencyChecks.[hc]pp
```
?  In that case we **might** need to find a new home for the other files in [`opm/core/props`](https://github.com/OPM/opm-simulators/tree/54d303ae5d40bb397a214fa45f808e4b3b94c4d8/opm/core/props), but that's probably a good idea too.

Reply:
that was the idea yeah. if the rest of the core directory suffers the same fate that  is also fine indeed.

Reply:
> Maybe we should take this opportunity to replace core with something more descriptive like 'diagnostics' or such

I have rebased this PR on top of #5440 which moves the contents of `opm/core/props` into `opm/simulators/utils` instead.